SE-14997: Adding Mule 4.2.2 workaround instructions (#216)
mulesoft/docs-api-manager: modules/ROOT/pages/proxy-domain-support.adoc

** `api-proxy-listener-http`: The listener configuration for HTTP communications 
** `api-proxy-listener-https`: The listener configuration for HTTPS communications
+

> . Copy the Maven artifact to the `domains` folder of your running Mule distribution, for example, `gateway-proxy-domain-1.0.0-mule-application-template.jar` and replace `mule-application-template` with `mule-domain`, resulting in the string: `gateway-proxy-domain-1.0.0-mule-domain.jar`.

Reviewer: ```suggestion
. Copy the Maven artifact to the `domains` folder of your running Mule distribution, for example, `gateway-proxy-domain-1.0.0-mule-application-template.jar`, and replace `mule-application-template` with `mule-domain`, resulting in the string `gateway-proxy-domain-1.0.0-mule-domain.jar`.
```
I think this is fine as one sentence, although it is complex. You could make it two steps, if you wanted.

Reviewer 2: I thought of it too, but then let it go since it was harder to link the two when trying to understand what needs to be done. I made into two steps now anyway.

<version>1.0.0</version>
<packaging>mule-domain</packaging>
----
[WARNING]

> For Mule (versions 4.2.2 or later) to be able to link to generated proxies, ensure that the domain file is named `proxy-shared-domain.jar` when copying it to the `domains` folder.

Reviewer: ```suggestion
For Mule 4.2.2 or later to link to generated proxies, the domain file must be named `proxy-shared-domain.jar` when copied to the `domains` folder.
```

Reviewer 2: Done! Thanks!

inbound.keystore.alias=alias
----

[WARNING]
====

> If you are using Mule 4.2.2 or later with the API Gateway Domain Template versions 1.0.x, you must modify the domain Java archive (JAR) file for Mule to link with the generated proxies:

Reviewer: https://docs.mulesoft.com/api-manager/2.x/proxy-domain-support:
```suggestion
If you are using Mule 4.2.2 or later with the API gateway domain template 1.0.x, you must modify the domain Java archive (JAR) file for Mule to link with the generated proxies:
```

Reviewer 2: Done! Thanks!

> * In `META-INF/maven/com.anypoint.mulesoft/gateway-proxy-domain/pom.xml`: change the value under `<groupId>` for `com.mulesoft.anypoint`.

Reviewer: ```suggestion
* In `META-INF/maven/com.anypoint.mulesoft/gateway-proxy-domain/pom.xml`, change the value of `groupId` to `com.mulesoft.anypoint`.
```
Please review carefully to ensure that this edit does not change your intended meaning.

Reviewer 2: Done! Thanks!

> * In `META-INF/mule-artifact/classloader-model.json`: change the value under `"groupId":` for `com.mulesoft.anypoint`.

Reviewer: ```suggestion
* In `META-INF/mule-artifact/classloader-model.json`, change the value of `groupID` to `com.mulesoft.anypoint`.
```
ditto

Reviewer 2: Done! Thanks!
